pkg/cloud/cloud.go

```diff
@@ -1527,20 +1531,27 @@ func (c *cloud) listSnapshots(ctx context.Context, request *ec2.DescribeSnapshot
 }
 
 // waitForVolume waits for volume to be in the "available" state.
-// On a random AWS account (shared among several developers) it took 4s on average.
 func (c *cloud) waitForVolume(ctx context.Context, volumeID string) error {
+	// Based on our testing in us-west-2 and ap-south-1, the median/p99 time until volume creation is ~1.5/~4 seconds.
+	// We have found that the following parameters are optimal for minimizing provisioning time and DescribeVolumes calls
+	// we queue DescribeVolume calls after [1.25, 0.5, 0.75, 1.125, 1.7, 2.5, 3.8, 5.7, 8.6, 12.8, 19.2] seconds.
+	// In total, we wait for ~60 seconds.
 	var (
-		checkInterval = 3 * time.Second
-		// This timeout can be "ovewritten" if the value returned by ctx.Deadline()
-		// comes sooner. That value comes from the external provisioner controller.
-		checkTimeout = 1 * time.Minute
+		initialDelay = volumeCreationPollInitialDelay
+		backoff      = wait.Backoff{
+			Duration: volumeCreationPollBackoffDuration,
+			Factor:   volumeCreationPollBackoffFactor,
+			Steps:    volumeCreationPollBackoffSteps,
+		}
 	)
 
+	time.Sleep(initialDelay)
+
 	request := &ec2.DescribeVolumesInput{
 		VolumeIds: []string{volumeID},
 	}
 
-	err := wait.PollUntilContextTimeout(ctx, checkInterval, checkTimeout, false, func(ctx context.Context) (done bool, err error) {
+	err := wait.ExponentialBackoffWithContext(ctx, backoff, func(ctx context.Context) (done bool, err error) {
```

Comment:
Suggestion: change the max backoff time to 3 seconds so that once the exponential backoff "ramps up", it behaves similarly to today

Reply:
> Notice a meaningful number of CreateVolume RPCs taking longer than 10 seconds with this change.

Could that be because we succeed sooner on rpcs and then get rate-limited for CreateVolume sooner with this PR?

We don't take longer than 3s for a backoff until the 7th call of createvolume. 
[1.25, 0.5, 0.75, 1.125, 1.7, 2.5, 3.8, 5.7, 8.6, 12.8, 19.2]

But yes agreed we can decrease the max backoff time.